buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/PerformanceTest.groovy

```diff
@@ -14,109 +14,104 @@
-import org.gradle.api.tasks.Input;
-import org.gradle.api.tasks.Optional;
-import org.gradle.api.tasks.OutputDirectory;
-import org.gradle.gradlebuild.test.integrationtests.DistributionTest;
+import groovy.transform.CompileStatic
+import org.gradle.api.internal.tasks.options.Option
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.gradlebuild.test.integrationtests.DistributionTest
 
-import javax.annotation.Nullable;
-import java.io.File;
+import javax.annotation.Nullable
 
 /**
  * A test that checks execution time and memory consumption.
  */
-public class PerformanceTest extends DistributionTest {
+@CacheableTask
+@CompileStatic
+class PerformanceTest extends DistributionTest {
+    @Nullable
+    String baselines
 
-    private String baselines;
-    private String warmups;
-    private String runs;
-    private String checks;
-    private String channel;
-    private File debugArtifactsDirectory = new File(getProject().getBuildDir(), getName());
+    @Nullable
+    String warmups
 
-    @Option(option = "scenarios", description = "A semicolon-separated list of performance test scenario ids to run.")
-    public void setScenarios(String scenarios) {
-        systemProperty("org.gradle.performance.scenarios", scenarios);
-    }
+    @Nullable
+    String runs
 
-    @Option(option = "baselines", description = "A comma or semicolon separated list of Gradle versions to be used as baselines for comparing.")
-    public void setBaselines(@Nullable String baselines) {
-        this.baselines = baselines;
-        systemProperty("org.gradle.performance.baselines", baselines);
-    }
+    @Nullable
+    String checks
 
-    @Nullable @Optional @Input
-    public String getBaselines() {
-        return baselines;
-    }
+    @Nullable
+    String channel
 
-    @Option(option = "warmups", description = "Number of warmups before measurements")
-    public void setWarmups(@Nullable String warmups) {
-        this.warmups = warmups;
-        systemProperty("org.gradle.performance.execution.warmups", warmups);
-    }
+    @Input
+    String resultStoreClass = "org.gradle.performance.results.AllResultsStore"
 
-    @Nullable @Optional @Input
-    public String getWarmups() {
-        return warmups;
-    }
+    @OutputDirectory
+    File debugArtifactsDirectory = new File(getProject().getBuildDir(), getName())
 
-    @Option(option = "runs", description = "Number of iterations of measurements")
-    public void setRuns(@Nullable String runs) {
-        this.runs = runs;
-        systemProperty("org.gradle.performance.execution.runs", runs);
-    }
+    @OutputDirectory
+    File reportDir
 
+    PerformanceTest() {
+        outputs.doNotCacheIf("last and nightly version don't need cache at all", {
+            List baselineList = baselines == null ? [] : baselines.split(',').collect { String baseline -> baseline.trim() }
```

Comment:
IIUC, the motivation for converting this class from `Java` to `Groovy` was that you can use `split` and `collect` here? Let's use `Splitter` from guava instead.